### PR TITLE
Support for defines declared in an external module

### PR DESCRIPTION
**Pull request by aowainati**
Same as this PR: #3 + an additional fix for crazy babel-ouroboros

### CODE REVIEW

File: modules/config.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+  if (!projectRoot) {
+    projectRoot = findProjectRoot();
+  }
+
+  const fullModulePath = path.resolve(projectRoot, modulePath);
+  let replacements = JSON.parse(
+    // Shell out before requiring the config module. Otherwise, you end up with a chicken-and-egg
+    // where babel is not yet fully initialized and the config module silently comes back empty
+    childProcess.execSync(`node -e "console.log(JSON.stringify(require('${fullModulePath}')))"`, {
```
Comment (nevir) [inline modules/config.js]: No backticks - let's keep compatibility with pre-v4 node

File: modules/config.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      encoding: 'utf-8',
+    })
+  );
+
+  // Support es6 style modules with default exports.
+  if (replacements.__esModule && replacements.default) {
+    replacements = replacements.default;
+  }
```
Comment (nevir) [inline modules/config.js]: This'll never occur now that we're JSON-encoding it

File: modules/config.js (hunk 1 above)
Comment (aowainati) [inline modules/config.js]: Done

File: modules/config.js (hunk 2 above)
Comment (aowainati) [inline modules/config.js]: Done

**Pull request merged**